### PR TITLE
Add missing Javadoc to all functions in GraphPath.java

### PR DESCRIPTION
**Pull request by HakanIlbas**
GraphPath.java in ```Ghidra/Framework/Graph/src/main/java/ghidra/graph``` did not have a lot of documentation for its function. As someone new who just started contributing to Ghidra, it seemed like an easy way to get more familiar with the project. This PR adds documentation to all of the functions in GraphPath.java by adding JavaDoc.

### CODE REVIEW

**Review by dragonmacher — COMMENTED** [review]
Overall the line wrapping is not quite up to our current coding standards.   We like lines to get as close as possible to 100 characters, without going over.  Some of your new comments wrap before this mark.

**Comment by HakanIlbas** [comment]
Oh okay i will be sure to solve that. I have looked around the repository regarding things like this, but i could not find anything. Do you have a file or something regarding code/documentation styling?

**Comment by HakanIlbas** [comment]
Solved the line wrapping inconsistencies. Is there anything else you would like to see getting improved?

File: Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java [hunk 1]
```diff
@@ -30,21 +30,41 @@
...
+	 * Default constructor.
+	 */
 	public GraphPath() {
-		// default constructor
 	}
 
+	/**
+	 * Constructor with a vertex.
+	 *
+	 * @param v being the first vertex of the newly initialized GraphPath object
```
Comment (dragonmacher) [inline Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java:42]: don't need the word 'being'

File: Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java [hunk 2]
```diff
@@ -30,21 +30,41 @@
...
+	 */
 	public GraphPath(V v) {
 		add(v);
 	}
 
+	/**
+	 * Creates a new GraphPath object by performing a deep copy
```
[the remark this answers is not in the document]
Comment (HakanIlbas) [reply Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java:49]: Oh okay my bad. I used the ```addAll``` function a while ago and tought that it created a deep copy, will fix this together with all other requested changes in the next commit.

File: Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java [hunk 3]
```diff
@@ -58,6 +78,15 @@ public boolean startsWith(GraphPath<V> otherPath) {
 		return true;
 	}
 
+	/**
+	 * Return the first few nodes that two GraphPaths have in common.
```
[the remark this answers is not in the document]
Comment (HakanIlbas) [reply Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java:82]: Thanks, also this should have been vertices instead of nodes

**Comment by HakanIlbas** [comment]
I think that all requested changes are now added

**Comment by astrelsky** [comment]
> Oh okay i will be sure to solve that. I have looked around the repository regarding things like this, but i could not find anything. Do you have a file or something regarding code/documentation styling?

In the Ghidra installation root, go to the docs folder and you will find the file `GhidraCodingStandards.html`.

File: Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java [hunk 4]
```diff
@@ -30,21 +30,41 @@
...
+	 */
 	public GraphPath<V> copy() {
 		GraphPath<V> newPath = new GraphPath<>();
 		newPath.pathList.addAll(pathList);
 		newPath.pathSet.addAll(pathSet);
 		return newPath;
 	}
 
+	/**
+	 * Check if a GraphPath starts with another Graphpath.
```
[the remark this answers is not in the document]
Comment (gbear14275) [reply Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java:61]:
```suggestion
	 * Check if a GraphPath starts with another GraphPath.
```

File: Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java [hunk 5]
```diff
@@ -30,21 +30,40 @@
...
+	 * @param v the first vertex of the newly initialized GraphPath object
+	 */
 	public GraphPath(V v) {
 		add(v);
 	}
 
+	/**
+	 * Creates a new GraphPath object by performing a shallow copy on another GraphPath object.
+	 *
+	 * @return the new deepcopy of the original GraphPath object
```
[the remark this answers is not in the document]
Comment (gbear14275) [reply Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java:51]:
```suggestion
	 * @return the new shallow copy of the original GraphPath object
```

File: Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java [hunk 6]
```diff
@@ -68,43 +95,79 @@ public boolean startsWith(GraphPath<V> otherPath) {
...
+	 * @return size of the GraphPath
+	 */
 	public int size() {
 		return pathList.size();
 	}
 
+	/**
+	 * Check if vertex v is in the GraphPath.
+	 *
+	 * @return true if vertex v is in the pathSet of the GraphPath
```
[the remark this answers is not in the document]
Comment (gbear14275) [reply Ghidra/Framework/Graph/src/main/java/ghidra/graph/GraphPath.java:110]:
```suggestion
	 * @return true if vertex v is in this GraphPath
```

**Comment by HakanIlbas** [comment]
Took a while, but the last few requested changes should have been fixed now

**Pull request closed without merging**